src/benchq/resource_estimation/graph_compilation.py

```diff
@@ -189,29 +239,151 @@ def get_resource_estimations_for_program(
         clifford_t_circuit = pyliqtr_transpile_to_clifford_t(
             circuit, synthesis_accuracy=synthesis_error_budget
         )
-        graph = get_algorithmic_graph(clifford_t_circuit)
-        n_nodes_list.append(len(graph))
+        graphs_list.append(get_algorithmic_graph(clifford_t_circuit, verbose))
+        with open("icm_output.json", "r") as f:
+            output_dict = json.load(f)
+            data_qubits_map = output_dict["data_qubits_map"]
+        data_qubits_map_list.append(data_qubits_map)
 
     return resource_estimations_for_subcomponents(
-        n_nodes_list,
+        graphs_list,
+        data_qubits_map_list,
         quantum_program,
         architecture_model,
         ec_error_budget,
+        use_full_program_graph=use_full_program_graph,
+        verbose=verbose,
     )
 
 
 def resource_estimations_for_subcomponents(
-    n_nodes_list, quantum_program, architecture_model, tolerable_circuit_error_rate
+    graphs_list,
+    data_qubits_map_list,
+    quantum_program,
+    architecture_model,
+    tolerable_circuit_error_rate,
+    use_full_program_graph: bool = False,
```

Comment:
It would be useful to get some explanation of why we need this param.

Reply:
Done!